sources/models/src/modeled_types/shared.rs

```diff
@@ -647,6 +647,127 @@ mod test_sysctl_key {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// BootConfigKey represents a string that is a valid Kernel boot config keyword; Each key word must
+/// contain only alphabets, numbers, dash (-) or underscore (_).
+/// BootConfigKey stores the original string and makes it accessible through standard traits.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct BootConfigKeyword {
+    inner: String,
+}
+
+impl TryFrom<&str> for BootConfigKeyword {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, error::Error> {
+        let valid_keyword = input
+            .chars()
+            .all(|c| (c.is_ascii() && c.is_alphanumeric()) || c == '-' || c == '_')
+            && !input.is_empty();
+        ensure!(
+            valid_keyword,
+            error::InvalidBootconfigKeywordSnafu { input }
+        );
+        Ok(BootConfigKeyword {
+            inner: input.to_string(),
+        })
+    }
+}
+
+string_impls_for!(BootConfigKeyword, "BootConfigKeyword");
+
+#[cfg(test)]
+mod test_bootconfig_keyword {
+    use super::BootConfigKeyword;
+    use std::convert::TryFrom;
+
+    #[test]
+    fn valid_bootconfig_keyword() {
+        for ok in &[
+            "keyword",
```

Comment:
Nit: I don't think the documentation explicitly disallows words that start with `-` or `_`, so let's add an extra test here to cover that.